source/codegen/proto_helpers.py

```diff
@@ -5,7 +5,8 @@ def should_allow_alias(enum):
   if enum.get("generate-mappings", False):
     return False
   enum_values = [e["value"] for e in enum["values"]]
-  return (0 in enum_values) or (len(enum_values) != len(set(enum_values)))
+  # aliases are allowed for zero-values if they are not first in the list.
+  return 0 in enum_values[1:] or len(enum_values) != len(set(enum_values))
```

Comment:
Do you think we should add metadata validation to make sure we don't get into the first condition with a regular (non-attribute) enum? I'm assuming for all regular integer enums in the metadata, we always expect the 0 value (if it's present) to be the first value.

If so, I played with
```python
if type(values[0]) is int:
                if 0 in values and values[0] != 0:
                    raise Exception(f"0 value in integer enum needs to be the first value!")
```
at the end of `validate_enum` in metadata_validation.py and it seemed like it worked.

Reply:
Yeah, I think that's a good idea. 

But looking harder at this it seems like we should be able to remove the `0 in ...` check. We're expecting that all aliases are now true aliases in the input value set. Not aliases introduced by adding `UNSPECIFIED` (since we explicitly don't want to do that).

Reply:
I'm now handling this in metadata_mutation. Also, if this ever sneaks in for an enum that is not attribute values, there will be an error from protoc. So I think this is covered now.

I also reverted the hand-edits.